pkg/relayer/proxy/relay_builders.go

```diff
@@ -10,42 +10,45 @@ import (
 
 // newRelayRequest builds a RelayRequest from an http.Request.
 func (sync *synchronousRPCServer) newRelayRequest(request *http.Request) (*types.RelayRequest, error) {
-	requestBz, err := io.ReadAll(request.Body)
+	requestBody, err := io.ReadAll(request.Body)
 	if err != nil {
 		return nil, err
 	}
+	defer request.Body.Close()
 
 	sync.logger.Debug().Msg("unmarshaling relay request")
 
 	var relayReq types.RelayRequest
-	if err := relayReq.Unmarshal(requestBz); err != nil {
+	if err := relayReq.Unmarshal(requestBody); err != nil {
 		sync.logger.Debug().Msg("unmarshaling relay request failed")
 		return nil, err
 	}
 
 	return &relayReq, nil
 }
 
-// newRelayResponse builds a RelayResponse from an http.Response and a SessionHeader.
+// newRelayResponse builds a RelayResponse from a response body reader and a SessionHeader.
 // It also signs the RelayResponse and assigns it to RelayResponse.Meta.SupplierSignature.
-// The response's Body is passed directly into the RelayResponse.Payload field.
+// The response body is passed directly into the RelayResponse.Payload field.
 func (sync *synchronousRPCServer) newRelayResponse(
-	response *http.Response,
+	responseBody io.ReadCloser,
 	sessionHeader *sessiontypes.SessionHeader,
 ) (*types.RelayResponse, error) {
+	defer responseBody.Close()
```

Comment:
Ditto. I don't think we shold be closing things inside of helpers. It creates side effects.